Mail/Utils/NavigationState.swift

```diff
@@ -92,28 +93,41 @@ class NavigationState: ObservableObject {
     /// The selected thread is the last in collection, by convention.
     @Published var threadPath = [Thread]()
 
+    private(set) var account: Account?
+
     init() {
-        if accountManager.currentAccount != nil,
-           let currentMailboxManager = accountManager.currentMailboxManager {
-            rootViewState = .mainView(currentMailboxManager)
-        } else if !accountManager.mailboxes.isEmpty && accountManager.mailboxes.allSatisfy({ !$0.isAvailable }) {
-            rootViewState = .unavailableMailboxes
-        } else {
-            rootViewState = .onboarding
-        }
+        account = AccountManager.instance.getCurrentAccount()
+        rootViewState = NavigationState.getMainViewStateIfPossible()
 
         accountManagerObservation = accountManager.objectWillChange.receive(on: RunLoop.main).sink { [weak self] in
-            self?.switchToCurrentMailboxManagerIfPossible()
+            self?.account = AccountManager.instance.getCurrentAccount()
+            self?.rootViewState = NavigationState.getMainViewStateIfPossible()
         }
     }
 
+    static func getMainViewStateIfPossible() -> RootViewState {
+        let accountManager = AccountManager.instance
+        if let currentAccount = accountManager.getCurrentAccount() {
```

Comment:
Could we write it with a guard statement ?

Reply:
FTR: Discussed IRL